src/System.Private.CoreLib/src/System/Resources/ResourceManager.cs

```diff
@@ -889,21 +889,21 @@ public virtual string GetString(string name, CultureInfo culture)
                 throw new ArgumentNullException(nameof(name));
 
 #if FEATURE_APPX
-            if (ApplicationModel.IsUap)
+            if (_bUsingModernResourceManagement)
             {
-                // If the caller explictily passed in a culture that was obtained by calling CultureInfo.CurrentUICulture,
-                // null it out, so that we re-compute it.  If we use modern resource lookup, we may end up getting a "better"
-                // match, since CultureInfo objects can't represent all the different languages the AppX resource model supports.
-                // For classic resources, this causes us to ignore the languages list and instead use the older Win32 behavior,
-                // which is the design choice we've made. (See the call a little later to GetCurrentUICultureNoAppX()).
-                if (object.ReferenceEquals(culture, CultureInfo.CurrentUICulture))
+                if (ApplicationModel.IsUap)
```

Comment:
The `ApplicationModel.IsUap` condition looks redundant. If `_bUsingModernResourceManagement` is set, `ApplicationModel.IsUap` must be set as well.